# 5.异常捕获和错误处理

## 中间层处理

```js
app.use(async (ctx, next) => {
	console.log('a');
	await next();
	throw new Error('hello'); // 在控制台打印 hello ， 属于框架级的处理
	console.log('b');
})
```

context.js 中 的 onerror 属于中间层处理

## 框架层处理

## 总结